build-aux/m4/ax_pthread.m4

```diff
@@ -294,7 +294,7 @@ if test "x$ax_pthread_clang" = "xyes"; then
 	     # step
 	     ax_pthread_save_ac_link="$ac_link"
 	     ax_pthread_sed='s/conftest\.\$ac_ext/conftest.$ac_objext/g'
-	     ax_pthread_link_step=`$as_echo "$ac_link" | sed "$ax_pthread_sed"`
+	     ax_pthread_link_step=`AS_ECHO(["$ac_link"]) | sed "$ax_pthread_sed"`
```

Comment:
@daira usually says that we should not make changes to the M4 files for licensing reasons. This issue has been fixed in the source M4 file, so we should update to that instead.

Upstream made that effective change across three PRs:
- https://github.com/bitcoin/bitcoin/pull/8198
- https://github.com/bitcoin/bitcoin/pull/19558
  - This PR also includes related build system changes.
- https://github.com/bitcoin/bitcoin/pull/23616